### PR TITLE
Fix Templates type hints

### LINKED ISSUE #989: Fix type hints to make use of new benchmark structure
I think we should update the typing in function signatures such as https://github.com/ContinualAI/avalanche/blob/master/avalanche/training/templates/base.py#L74 requiring specific experiences to be more general, replacing them with the new [CLExperience](https://github.com/ContinualAI/avalanche/blob/master/avalanche/benchmarks/scenarios/generic_scenario.py#L73), perhaps even swapping the `Sequence` of experiences requirement with a [CLStream](https://github.com/ContinualAI/avalanche/blob/master/avalanche/benchmarks/scenarios/generic_scenario.py#L154).

### PR DESCRIPTION
**Pull request by NickLucche**
First draft attempting to fix #989

### CODE REVIEW

File: avalanche/training/templates/base.py [hunk 1]
```diff
@@ -52,15 +54,13 @@ def __init__(
...
+        self.experience: Optional[CLExperience] = None
         """ Current experience. """
 
         self.is_training: bool = False
         """ True if the strategy is in training mode. """
 
-        self.current_eval_stream: Optional[
-            Sequence[ClassificationExperience]
-        ] = None
+        self.current_eval_stream: Optional[CLStream] = None
```
Comment (AntonioCarta) [inline avalanche/training/templates/base.py:63]: this is taken from the train and eval arguments, so it should be an `ExpSequence`

**Review by AntonioCarta — CHANGES_REQUESTED** [review]

**Comment by AntonioCarta** [comment]
Thanks, just a minor comment and then we are ready to merge!

File: avalanche/training/templates/base.py (hunk 1 above)
Comment (NickLucche) [reply avalanche/training/templates/base.py:63]: nice one, I'll fix it right up

**Comment by NickLucche** [comment]
nice

**Pull request merged**